rails_docker/Dockerfile

```diff
@@ -12,78 +12,95 @@ ENV RACK_ENV=$RUBY_ENV \
     LC_ALL="en_US.UTF-8" \
     LANGUAGE="en_US:en"
 
-RUN apt-get update -qq && \
-    apt-get install -y --no-install-recommends apt-transport-https curl gnupg
-
-# Add Yarn repository
-ADD https://dl.yarnpkg.com/debian/pubkey.gpg /tmp/yarn-pubkey.gpg
-RUN apt-key add /tmp/yarn-pubkey.gpg && rm /tmp/yarn-pubkey.gpg && \
-    echo "deb http://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list
-
-# Add the PPA (personal package archive) maintained by NodeSource
-# This will have more up-to-date versions of Node.js than the official Debian repositories
-RUN curl -sL https://deb.nodesource.com/setup_"$NODE_VERSION".x | bash -
-
-# Install general required core packages, Node JS related packages and Chrome (testing)
-RUN apt-get update -qq && \
-    apt-get install -y --no-install-recommends build-essential libpq-dev nodejs yarn && \
-    apt-get install -y --no-install-recommends rsync locales chrpath pkg-config libfreetype6 libfontconfig1 git cmake wget unzip && \
-    apt-get clean && \
+# Install dependencies
+RUN apk add \
+  --no-cache \
+  --update build-base \
+    build-base \
+    linux-headers \
+    bash \
+    make \
+    cmake \
+    imagemagick \
+    ruby \
+    ruby-bundler \
+    ruby-irb \
+    libxml2-dev \
+    libxslt-dev \
+    libffi-dev \
+    postgresql-dev \
+    zlib-dev \
+    git \
+    nodejs-current \
+    nodejs-npm \
+    tzdata && \
     rm -rf /var/lib/apt/lists/*
 
-# Set up the Chrome PPA and install Chrome Headless
+# Installs Chromium package from edge.
 RUN if [ "$BUILD_ENV" = "test" ]; then \
-      wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
-      echo 'deb http://dl.google.com/linux/chrome/deb/ stable main' >> /etc/apt/sources.list.d/google-chrome.list && \
-      apt-get update -qq && \
-      apt-get install -y --no-install-recommends google-chrome-stable && \
-      rm /etc/apt/sources.list.d/google-chrome.list && \
-      apt-get clean && \
-      rm -rf /var/lib/apt/lists/* ; \
-    fi
-
-RUN sed -i "s/^#\ \+\(en_US.UTF-8\)/\1/" /etc/locale.gen
-RUN locale-gen en_US.UTF-8
+  echo @edge http://nl.alpinelinux.org/alpine/edge/community >> /etc/apk/repositories \
+    && echo @edge http://nl.alpinelinux.org/alpine/edge/main >> /etc/apk/repositories \
+    && apk add --no-cache \
+    chromium@edge \
+    harfbuzz@edge \
+    nss@edge \
+    freetype@edge \
+    ttf-freefont@edge && \
+    rm -rf /var/lib/apt/lists/* ; \
+fi
 
 RUN mkdir "$APP_HOME"
+
 # Replace by the following if the application uses Rails engines
 # RUN mkdir "$APP_HOME" "$APP_HOME/engines"
 WORKDIR $APP_HOME
 
-# Only copy the dependency definition files (Gemfile and packages) to use Docker cache for these steps
-# Install Ruby dependencies
-COPY Gemfile* ./
-# Copy each engine lib/, Gemfile and .gemspec files
-# Example:
-# COPY engines/app_auth/lib/ ./engines/app_auth/lib/
-# COPY engines/app_auth/Gemfile engines/app_auth/app_auth.gemspec ./engines/app_auth/
```

Comment:
We need to retain the comment about copying the engines files as all apps we build use engines

Reply:
done